File: solved_class3/11723.py
import sys
input = sys.stdin.readline

s = set()
testCase = int(input().strip())

for _ in range(testCase):
    data = input().rstrip()
    try:
        operator, x = data.split()
        x = int(x)
    except:
        operator = data
    if operator == "add":
        s.add(x)
    elif operator == "remove":
        s.discard(x)
    elif operator == "check":
        print(1) if x in s else print(0)
    elif operator == "toggle":
        s.remove(x) if x in s else s.add(x)
    elif operator == "all":
        s = set(range(1, 21))
    elif operator == "empty":
        s.clear()

File: solved_class3/11723.cpp
#include <iostream>
using namespace std;

int main()
{
    cin.tie(0)->sync_with_stdio(0);

    int testCase, s = 0;
    cin >> testCase;

    while (testCase--)
    {
        string op;
        int x;
        cin >> op;

        if (op != "all" && op != "empty")
            cin >> x;

        if (op == "add")
            s |= 1 << x;
        else if (op == "remove")
            s &= ~(1 << x);
        else if (op == "check")
        {
            if (s & (1 << x))
                cout << "1\n";
            else
                cout << "0\n";
        }
        else if (op == "toggle")
            s ^= 1 << x;
        else if (op == "all")
            s = (1 << 21) - 1;
        else if (op == "empty")
            s = 0;
    }

    return 0;
}
